File: tests/fixtures/code/sample.py
"""Sample Python module for testing symbol search."""


def hello_world():
    """Greet the world."""
    return "Hello, World!"


def calculate_sum(a: int, b: int) -> int:
    """Add two numbers."""
    return a + b


class SampleClass:
    """A sample class with methods."""

    def __init__(self, name: str):
        self.name = name

    def method_one(self) -> str:
        """First method."""
        return f"Hello, {self.name}"

    def method_two(self, value: int) -> int:
        """Second method."""
        return value * 2


GLOBAL_CONSTANT = 42

File: tests/fixtures/code/sample.rs
//! Sample Rust module for testing symbol search.

/// Greet someone by name.
pub fn greet(name: &str) -> String {
    format!("Hello, {}!", name)
}

/// A sample struct with methods.
pub struct Greeter {
    pub name: String,
}

impl Greeter {
    /// Create a new greeter.
    pub fn new(name: &str) -> Self {
        Self {
            name: name.to_string(),
        }
    }

    /// Say hello.
    pub fn say_hello(&self) -> String {
        greet(&self.name)
    }
}

/// A constant value.
pub const SAMPLE_CONSTANT: u32 = 100;
